core/src/main/java/org/elasticsearch/gateway/TransportNodesListGatewayStartedShards.java

```diff
@@ -261,29 +272,40 @@ public String getIndexUUID() {
     public static class NodeGatewayStartedShards extends BaseNodeResponse {
 
         private long version = -1;
+        private String exception = null;
 
         NodeGatewayStartedShards() {
         }
-
         public NodeGatewayStartedShards(DiscoveryNode node, long version) {
+            this(node, version, null);
+        }
+
+        public NodeGatewayStartedShards(DiscoveryNode node, long version, String exception) {
             super(node);
             this.version = version;
+            this.exception = exception;
         }
 
         public long version() {
             return this.version;
         }
 
+        public String exception() {
+            return this.exception;
+        }
+
         @Override
         public void readFrom(StreamInput in) throws IOException {
             super.readFrom(in);
             version = in.readLong();
+            exception = in.readOptionalString();
         }
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
             super.writeTo(out);
             out.writeLong(version);
+            out.writeOptionalString(exception);
```

Comment:
you can actually serialize the entire exception with `StreamOutput#writeThrowable` it might be better to have the stacktrace too?